src/Serilog.Sinks.Splunk/Sinks/Splunk/SplunkJsonFormatter.cs

```diff
@@ -10,56 +10,140 @@
-        /// <param name="template"></param>
-        /// <param name="delim"></param>
-        /// <param name="output"></param>
-        protected override void WriteMessageTemplate(string template, ref string delim, TextWriter output)
+        /// <param name="formatProvider">Supplies culture-specific formatting information, or null.</param>
+        /// <param name="renderTemplate">If true, the template used will be rendered and written to the output as a property named MessageTemplate</param>
+        /// <param name="index">The Splunk index to log to</param>
+        /// <param name="source">The source of the event</param>
+        /// <param name="sourceType">The source type of the event</param>
+        /// <param name="host">The host of the event</param>
+        public SplunkJsonFormatter(
+            bool renderTemplate,
+            IFormatProvider formatProvider,
+            string source,
+            string sourceType,
+            string host,
+            string index)
         {
-            if(_renderTemplate)
-                base.WriteMessageTemplate(template, ref delim, output);
+            _renderTemplate = renderTemplate;
+            _formatProvider = formatProvider;
+
+            var suffixWriter = new StringWriter();
+            suffixWriter.Write("}"); // Terminates "event"
+
+            if (!string.IsNullOrWhiteSpace(source))
+            {
+                suffixWriter.Write(",\"source\":");
+                JsonValueFormatter.WriteQuotedJsonString(source, suffixWriter);
+            }
+
+            if (!string.IsNullOrWhiteSpace(sourceType))
+            {
+                suffixWriter.Write(",\"sourceType\":");
+                JsonValueFormatter.WriteQuotedJsonString(sourceType, suffixWriter);
+            }
+
+            if (!string.IsNullOrWhiteSpace(host))
+            {
+                suffixWriter.Write(",\"host\":");
+                JsonValueFormatter.WriteQuotedJsonString(host, suffixWriter);
+            }
+
+            if (!string.IsNullOrWhiteSpace(index))
+            {
+                suffixWriter.Write(",\"index\":");
+                JsonValueFormatter.WriteQuotedJsonString(index, suffixWriter);
+            }
+            suffixWriter.Write('}'); // Terminates the payload
+            _suffix = suffixWriter.ToString();
+        }
+
+        /// <inheritdoc/>
+        public void Format(LogEvent logEvent, TextWriter output)
+        {
+            if (logEvent == null) throw new ArgumentNullException(nameof(logEvent));
+            if (output == null) throw new ArgumentNullException(nameof(output));
+
+            output.Write("{\"time\":\"");
+            output.Write(logEvent.Timestamp.ToEpoch().ToString(CultureInfo.InvariantCulture));
+            output.Write("\",\"event\":{\"Level\":\"");
+            output.Write(logEvent.Level);
+            output.Write('"');
+
+            if (_renderTemplate)
+            {
+                output.Write("\",\"MessageTemplate\":");
```

Comment:
This one snuck in again `output.Write("\",\"MessageTemplate\":");`

Should be `output.Write(",\"MessageTemplate\":");`